cloudbridge/cloud/providers/gce/services.py

```diff
@@ -402,10 +422,133 @@ def list(self, limit=None, marker=None):
+                                .execute()
+
+    def get(self, instance_name):
+        """
+        Returns an instance given its name. Returns None
+        if the object does not exist.
+
+        GCE client API only supports returning the specified Instance resource
+        by its name (not by its id).
+        """
+        try:
+            response = self.provider.gce_compute\
+                               .instances() \
+                               .get(project=self.provider.project_name,
+                                    zone=self.provider.default_zone,
+                                    instance=instance_name) \
+                               .execute()
+            if response:
+                return GCEInstance(self.provider, response)
+        except googleapiclient.errors.HttpError as http_error:
+            # If the instance is not found, the API will raise
+            # googleapiclient.errors.HttpError.
+            cb.log.warning(
+                "googleapiclient.errors.HttpError: {0}".format(http_error))
+        return None
+
+    def find(self, name, limit=None, marker=None):
+        """
+        Searches for instances by instance name.
+        :return: a list of Instance objects
+        """
+        instances = [instance for instance in self.list()
+                     if instance.name == name]
+        if limit and len(instances) > limit:
+            instances = instances[:limit]
+        return instances
+
+    def list(self, limit=None, marker=None):
+        """
+        List all instances.
+        """
+        # For GCE API, Acceptable values are 0 to 500, inclusive.
+        # (Default: 500). If the number of available results is larger
+        # than maxResults, Compute Engine returns a nextPageToken that
+        # can be used to get the next page of results in subsequent
+        # list requests.
+        if limit is None or limit > 500:
+            max_result = 500
+        else:
+            max_result = limit
+        request = self.provider.gce_compute \
+                               .instances() \
+                               .list(project=self.provider.project_name,
+                                     zone=self.provider.default_zone,
+                                     maxResults=max_result)
+        instances = []
+        while request is not None:
+            response = request.execute()
+            instances.extend([GCEInstance(self.provider, inst)
+                              for inst in response['items']])
+            if limit and len(instances) >= limit:
+                break
+            request = self.provider.gce_compute \
+                                   .instances() \
+                                   .list_next(previous_request=request,
+                                              previous_response=response)
+        if limit and len(instances) > limit:
+            instances = instances[:limit]
+        return instances
```

Comment:
@baizhang There's a small interface mismatch here. The list method should return a ResultList object, instead of a List object. In this case, this can probably be wrapped in a ServerPagedResultList object.

Combined with the comment above, I think this function could possibly be rewritten as:
```
        max_result = limit if limit < 500 else 500
        response = self.provider.gce_compute.instances().list(project=self.provider.project_name,
                                     zone=self.provider.default_zone,
                                     maxResults=max_result,
                                     pageToken=marker)
        instances = [GCEInstance(self.provider, inst)
                             for inst in response['items']]
        return ServerPagedResultList(len(instances) > max_result,
                                     response['nextPageToken'],
                                     False, data=instances)
```